mozilla-release/toolkit/components/extensions/webrequest/WebRequest.jsm

```diff
@@ -737,8 +756,21 @@ HttpObserverManager = {
 
       if (isSecondaryTab) {
         gBrowser.removeTab(isSecondaryTab);
+      } else {
+        try {
```

Comment:
@tsl143 Please add our comment tag with the browser version.

Reply:
And also the same question about try/catch. Do you really need this?

Reply:
For try catch, I am expecting always the `tab[0]` exist, in weirdest case or by any future release this may change.

Reply:
as above

Reply:
> For try catch, I am expecting always the `tab[0]` exist, in weirdest case or by any future release this may change.

Then why not to check just its' existence? 
if (tabs[0] != null) {
  // here goes your code if tabs[0] exists;
}

Reply:
yeah, true, but I am just following try catch scheme here for different checks, so sticking to it.

Reply:
Ok. Then how can we know about the error? It will be silently ignored and swallowed by catch.
Either throw it explicitly in catch or use that tabs[0] against null check please.
I really do not want to fall into situation similar to what we had in SessionStore again.

Reply:
We have added console to catch block, for better error deduction.